.github/workflows/wait_for_required_workflows.yml

```diff
@@ -0,0 +1,73 @@
+name: Wait for all required workflows to pass
+
+on: [pull_request, push]
+
+jobs:
+  wait_for_required_workflows:
+    name: wait-for-required-workflows
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 2 # OR "2" -> To retrieve the preceding commit.
+      - name: Get changed files
+        id: changed-files
+        uses: tj-actions/changed-files@v31
+        with:
+          dir_names: true
+      - uses: actions/github-script@v6
+        env:
+          DIRS: ${{ steps.changed-files.outputs.all_changed_files }}
+          REF: ${{ github.ref }}
+        with:
+          script: |
+            const { REF, DIRS } = process.env
+            let now = new Date().getTime()
+            const deadline = now + 50 * 1000
+            let actions = ["cli",
```

Comment:
How about we keep the resolve modules logic we had to get this list?
That should ensure that if someone adds a new module, they are required to add a matching workflow file?

Looks like we can do it with https://github.com/featherweight-design/collect-directory-names-by-file-glob, WDYT?

Also I think we might also want to wait for the checks if someone changes the `source_test.yml` file for example, but I would keep that for another PR as it can complicate this script.

Reply:
Regarding the logic that sources that automatically let's do it in a follow-up PR and check that it all works now